workshops/web_chat/README.md

```diff
@@ -0,0 +1,782 @@
+            text-align: center;
+            word-wrap: break-word;
+        }
+
+        #message {
+            height: 100%;
+            width: 100%;
+            border: 1px solid gray;
+            border-radius: 5px;
+        }
+
+        #send {
+            height: 100%;
+            width: 100%;
+            border-radius: 5px;
+        }
+    </style>
+</head>
+
+<body>
+    <div id="messageLog"></div>
+    <div id="controls">
+        <input id="name" placeholder="Your name">
+        <textarea id="message" placeholder="Type your message here"></textarea>
+        <button id="send">
+            Send
+        </button>
+    </div>
+</body>
+</html>
+```
+
+</details>
+
+### Adding Javascript
+
+- Create a new file called frontend.js - This will be the code that runs on someone's device when they visit your webpage
+- Add a `<script src="frontend.js"></script>` just before body ends
+
+<details>
+    <summary> This is how your HTML file should look now: </summary>
+
+```html
+<!DOCTYPE html>
+<html>
+    
+<head>
+    <style>
+        * {
+            box-sizing: border-box;
+            font-family: sans-serif;
+            padding: 5px;
+            margin: 0px;
+            border: 0px;
+        }
+
+        b {
+            padding: 0px;
+        }
+
+        body,
+        html {
+            width: 100%;
+            height: 100%;
+        }
+
+        #messageLog {
+            width: 100%;
+            height: 90%;
+            overflow: auto;
+        }
+    
+        #controls {
+            width: 100%;
+            height: 10%;
+            display: grid;
+            grid-template-columns: 2fr 7fr 1fr;
+            grid-gap: 10px;
+        }
+
+        #name {
+            height: 100%;
+            width: 100%;
+            border: 1px solid gray;
+            border-radius: 5px;
+            text-align: center;
+            word-wrap: break-word;
+        }
+
+        #message {
+            height: 100%;
+            width: 100%;
+            border: 1px solid gray;
+            border-radius: 5px;
+        }
+
+        #send {
+            height: 100%;
+            width: 100%;
+            border-radius: 5px;
+        }
+    </style>
+</head>
+    
+<body>
+    <div id="messageLog"></div>
+    <div id="controls">
+        <input id="name" placeholder="Your name">
+        <textarea id="message" placeholder="Type your message here"></textarea>
+        <button id="send">
+            Send
+        </button>
+    </div>
+    <script src="frontend.js"></script>
+</body>
+</html>
+```
+
+</details>
+
+Congratulations on making it so far!
+
+![Hermione, Ron and Seamus cheering](https://cloud-pq5lbfiab.vercel.app/1cheer2.gif)
+
+## Programming the server (and frontend Javascript)
+
+> As (at the time of writing) Deno is still in beta @repl.it, you'll want to ignore any errors regarding import, await, and the Deno object.
+
+Let's get started with the server!
+
+- Head over to the index.ts file and input the following code
```

Comment:
Generally you should put file names between backticks. So index.ts should look like `index.ts`